flatgeobuf/index.qmd

```diff
@@ -0,0 +1,95 @@
+4. Last come the individual features. The rest of the file is a sequence of feature records, placed end to end in a row-wise fashion.
+
+### Row based
+
+Internally, features are laid out in a row-oriented fashion rather than a column-oriented fasion. This means that it's relatively cheap to select specific records from the file, but relatively expensive to select a specific column. This is ideal for a small spatial query (assuming an index exists in the file) but to load all geometries requires loading all attribute information as well.
+
+### No internal compression
+
+FlatGeobuf does not support compression while maintaining the ability to seek within the file. In particular, FlatGeobuf's spatial index describes the byte ranges in the _uncompressed_ file. Those byte ranges will be incorrect if the file is compressed.
+
+A compression like gzip can be applied to the FlatGeobuf file in full, but keep in mind that storing the compressed file will eliminate random access support.
+
+### No append support
+
+FlatGeobuf is a write-only format, and doesn't support appending, as that would invalidate the spatial index in the file.
+
+### Random access supported via spatial index
+
+FlatGeobuf optionally supports a spatial index at the beginning of the file, which can speed up reading portions of a file based on a spatial query. For more information on how this spatial index works, refer to the [Hilbert R Tree](./hilbert-r-tree.qmd) page.
+
+<!--
+TODO: Not sure whether this section makes sense to include
+
+### No internal chunking
+
+FlatGeobuf has no internal chunking. Rather, the binary features are stored sequentially with no other grouping.
+
+One implication of this is that you have
+
+And because FlatGeobuf is a row-based format, if some properties are really large, like if a column is a string description with many characters, you'll have a tradeoff between fetching lots of extra data to combine multiple requests into one or making many requests for small features that are scattered throughout the file. -->
+
+### Streaming features is supported
+
+FlatGeobuf supports _streaming_, meaning that you can use part of the file before the entire file has finished downloading. This is different than random access, because you have no ability to skip around in the file.
+
+Streaming can be valuable because it makes an application seem more responsive; you can have _something_ happen without having to wait for the full download to complete. A good example of this is [this example](https://observablehq.com/@bjornharrtell/streaming-flatgeobuf) by FlatGeobuf's author Björn Harrtell. As the file is downloaded to the browser, portions of the file get rendered progressively in parts.
+
+This works even with full-file compression like gzip or deflate because those compression algorithms support streaming decompression.
+
+### Broad type system
+
+FlatGeobuf supports attributes with a range of types:
```

Comment:
Did this list come from https://github.com/flatgeobuf/flatgeobuf/blob/master/src/net/FlatGeobuf/ColumnType.cs? Wondering if we could link to the source of this list so we don't have to include it here, especially in case it changes.

Reply:
Yeah, the definitive source is actually here https://github.com/flatgeobuf/flatgeobuf/blob/b013ac326568914aee43bcf3aa3c77574d629aa5/src/fbs/header.fbs#L26-L42.

I added a link in #22 but still listed the types here. I don't think the supported types in the spec are likely to change, because that would require an update to every implementation